src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -131,68 +132,74 @@ private void WriteRuntimeConfig(ProjectContext projectContext)
 
         private void AddFrameworks(RuntimeOptions runtimeOptions, ProjectContext projectContext)
         {
+            runtimeOptions.Tfm = TargetFramework;
+
+            List<RuntimeConfigFramework> frameworks = new List<RuntimeConfigFramework>();
+            if (projectContext.RuntimeFrameworks == null || projectContext.RuntimeFrameworks.Length == 0)
+            {
+                //  If there are no RuntimeFrameworks (which would be set in the ProcessFrameworkReferences task based
+                //  on FrameworkReference items), then use package resolved from MicrosoftNETPlatformLibrary for
+                //  the runtimeconfig
+                RuntimeConfigFramework framework = new RuntimeConfigFramework();
+                framework.Name = projectContext.PlatformLibrary.Name;
+                framework.Version = projectContext.PlatformLibrary.Version.ToNormalizedString();
+
+                frameworks.Add(framework);
+            }
+            else
+            {
+                HashSet<string> usedFrameworkNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                foreach (var platformLibrary in projectContext.RuntimeFrameworks)
+                {
+                    if (projectContext.RuntimeFrameworks.Length > 1 &&
+                        platformLibrary.Name.Equals("Microsoft.NETCore.App", StringComparison.OrdinalIgnoreCase) &&
+                        projectContext.IsFrameworkDependent)
+                    {
+                        //  If there are multiple runtime frameworks, then exclude Microsoft.NETCore.App,
+                        //  as a workaround for https://github.com/dotnet/core-setup/issues/4947
+                        //  The workaround only applies to normal framework references, included frameworks
+                        //  (in self-contained apps) must list all frameworks.
```

Comment:
That bug is fixed, right? Should we start listing all frameworks always?

Reply:
I wanted to keep the change to the minimum for 3.1. For 5.0 we should do that.